### PR TITLE
Fix Issue 7006 - Enabled pow to accept negative exponents for integrals

### PR DESCRIPTION
**Pull request by cristiancreteanu**

### CODE REVIEW

**Comment by ibuclaw** [comment]
Why should negative exponents be allowed when you can't represent any of them as an integer?

File: std/math.d [hunk 1]
```diff
@@ -7525,10 +7525,13 @@ if (isFloatingPoint!(F) && isIntegral!(G))
 typeof(Unqual!(F).init * Unqual!(G).init) pow(F, G)(F x, G n) @nogc @trusted pure nothrow
 if (isIntegral!(F) && isIntegral!(G))
 {
-    if (n<0) return x/0; // Only support positive powers
```
Comment (n8sh) [inline std/math.d]: Why does this runtime check exist instead of requiring the exponent to be unsigned?

Comment (dukc) [reply std/math.d]: This is a good reason to: https://forum.dlang.org/post/qobojhyjmxzvxorlhhnj@forum.dlang.org

File: std/math.d [hunk 2]
```diff
@@ -7525,10 +7525,13 @@ if (isFloatingPoint!(F) && isIntegral!(G))
 typeof(Unqual!(F).init * Unqual!(G).init) pow(F, G)(F x, G n) @nogc @trusted pure nothrow
```
Comment (joakim-noah) [inline std/math.d]: The documentation above will need to be corrected too, if this is to be accepted.

**Comment by berni44** [comment]
@RazvanN7 IMHO, this one can be closed - the bug has been fixed in #7322

**Comment by RazvanN7** [comment]
@berni44 Thanks for the heads up.

**Pull request closed without merging**